hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieActiveTimeline.java

```diff
@@ -433,6 +494,32 @@ private void createFileInPath(Path fullPath, Option<byte[]> content) {
     }
   }
 
+  /**
+   * Creates a new file in timeline with overwrite set to false. This ensures
+   * files are created only once and never rewritten
+   * @param fullPath File Path
+   * @param content Content to be stored
+   */
+  private void writeFileOnceInPath(Path fullPath, Option<byte[]> content) {
```

Comment:
rename to `createImmutableFileInPath` ?

Reply:
In other words why can't we use this method for the nullVersion too?

Reply:
@vinothchandar : Currently, we employ overwrite semantics for inflight commit/delta-commit files. We first create an empty inflight file and then overwrite it later with workload profile data. I can try cleaning that up in hoodie write client.  Let me know your thoughts ?

@n3nash : Renamed the method.